src/execution_impl.rs

```diff
@@ -83,21 +83,16 @@ pub(crate) fn exec(
     param: &str,
     interface: &dyn Interface,
 ) -> Result<Response> {
+    let mut env = Env::new(interface);
     let instance = match instance {
         Some(instance) => instance,
-        None => create_instance(limit, module, interface)?,
+        None => create_instance(limit, module, env.clone())?,
     };
-    let mut env = as_ffi_bindings::Env::default();
-    env.init(&instance)?;
+    env.init_with_instance(&instance)?;
 
     // Closure for the execution allowing us to handle a gas error
-    fn execution(
-        instance: &Instance,
-        function: &str,
-        param: &str,
-        env: &as_ffi_bindings::Env,
-    ) -> Result<Response> {
-        let param_ptr = *StringPtr::alloc(&param.to_string(), env)?;
+    fn execution(instance: &Instance, function: &str, param: &str, env: &Env) -> Result<Response> {
+        let param_ptr = *StringPtr::alloc(&param.to_string(), &env.wasm_env)?;
```

Comment:
right, so you can just use the `Env` created inside `create_instance` I would say.